app/controllers/api/v1/dependencies_controller.rb

```diff
@@ -0,0 +1,22 @@
+class Api::V1::DependenciesController < Api::BaseController
+  skip_before_action :verify_authenticity_token
+  GEM_REQUEST_LIMIT = 250
+
+  def index
+    return render text: '' if gem_names.empty?
+    return render text: "Too many gems!", status: 422 if gem_names.size > GEM_REQUEST_LIMIT
```

Comment:
I think this will work but I'm not sure if it's the best way to do it:
```ruby
    if gem_names.size > GEM_REQUEST_LIMIT
      respond_to do |format|
        format.json { render json: { error: "Too many gems (use --full-index instead)", code: 422 }, status: 422 }
        format.any { render text: "Too many gems!", status: 422, content_type: "text/plain" }
      end
      return
    end
```